Pr dsub notebook demo

johntbates opened the pull request:
* Added `dsub_notebook.py`, demonstration notebook to programmatic notebook execution dir.
* Updated `README.md` with instructions for executing demonstration workspace with `dsub_notebook.py`
* TODO: Instructions for running notebook with `notebook_workflow.wdl`.

deflaux commented on programmatic_execution_of_notebooks/README.md:
```diff
@@ -29,4 +29,82 @@ Details and limitations:
\ No newline at end of file
+* It is not compatible with notebooks written to run on [Hail](https://hail.is/) clusters.
+
+## [dsub_notebook.py][./dsub_notebook.py]
```
link syntax needs a fix

deflaux commented on programmatic_execution_of_notebooks/README.md:
```diff
@@ -29,4 +29,82 @@ Details and limitations:
+
+## [dsub_notebook.py][./dsub_notebook.py]
+
+Use `dsub_notebook.py` to xecute Jupyter notebooks from the Terra Workspace terminal command line.  As with `notebook.wdl`,
```
typo: xecute

deflaux commented on programmatic_execution_of_notebooks/README.md:
```diff
@@ -29,4 +29,82 @@ Details and limitations:
+This script will:
+* Optionally install a list of Python packages before executing the notebook.
+  * This is because a kernel restart is often necessary to make use of Python packages installed during notebook execution time.
+  * For R package dependencies, have the notebook install them at the beginning.
```
Here and above in the WDL section, can you please add:
> For R package dependencies, have the notebook install them at the beginning of the notebook.

deflaux commented on programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb:
```diff
@@ -0,0 +1,1240 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Use Papermill Parameters to Customize sample-level QC results\n",
```
This notebook should stay in https://github.com/verilylifesciences/variant-qc/blob/main/notebooks/ExploreSampleQCResults.ipynb and just receive these nice updates. I do see the value in having a test notebook along side these two approaches, but I think more of a 'hello world' notebook will do the trick as long as it emits a CSV or a PNG.

We can have two (or more) notebooks in the Terra featured workspace and they do not need to all come from the same git repo.

deflaux commented on programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb:
```diff
@@ -0,0 +1,1240 @@
+   "source": [
+    "retrieve_simons_sample_info <- function() {\n",
+    "    return(bq_table_download(bq_project_query(BILLING_PROJECT_ID, '\n",
+    "    SELECT\n",
```
nit: consider separating the constants (the queries) from the code so that there is no need to define these functions.

deflaux commented on programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb:
```diff
@@ -0,0 +1,1240 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Copyright 2022 Verily Life Sciences LLC. All rights reserved.\n",
```
The copyright should remain the same. This is just an update to a preexisting notebook.

deflaux commented on programmatic_execution_of_notebooks/README.md:
```diff
@@ -29,4 +29,82 @@ Details and limitations:
+
+## Demonstration notebook [DemoQCResultsWithParameters.ipynb](./DemoQCResultsWithParameters.ipynb)
+
+We have included a notebook to demonstrate how to programmatically execute a notebook using  `notebook.wdl` or
```
These instructions are great. I think they could be refactored in a follow up PR in to a notebook that someone could run to setup dsub and the script for the more interesting example with sample QC. And then these instructions remaining in the README could be updated for the 'hello world' notebook to be included here.

deflaux commented on programmatic_execution_of_notebooks/DemoQCResultsWithParameters.ipynb:
```diff
@@ -0,0 +1,1240 @@
+    "    ')))\n",
+    "}\n",
+    "\n",
+    "SAMPLE_INFO_FNS <- list(simons = retrieve_simons_sample_info,\n",
```
It would be great to move all the constants to a cell towards the top instead of mixing them in with the logic.

All in one cell:
```
CHROM_RESULTS_PATHS <- list(simons = 'gs://genomics-public-data/simons-genome-diversity-project/reports/Simons_Genome_Diversity_Project_sample_reference_results.csv',
                            platinum_genomes = 'gs://genomics-public-data/platinum-genomes/reports/DeepVariant_Platinum_Genomes_sample_reference_results.csv',
                            thousand_genomes = 'gs://genomics-public-data/1000-genomes-phase-3/reports/1000_Genomes_phase_3_sample_reference_results.csv')

GENOME_RESULTS_PATHS <- list(simons = 'gs://genomics-public-data/simons-genome-diversity-project/reports/Simons_Genome_Diversity_Project_sample_results.csv',
                             platinum_genomes = 'gs://genomics-public-data/platinum-genomes/reports/DeepVariant_Platinum_Genomes_sample_results.csv',
                             thousand_genomes = 'gs://genomics-public-data/1000-genomes-phase-3/reports/1000_Genomes_phase_3_sample_results.csv')


simons_sample_info_query <- '
    SELECT
      id_from_vcf AS name,
      "Illumina" AS sequencing_platform,
      sex,
      region AS ancestry
    FROM
      `bigquery-public-data.human_genome_variants.simons_genome_diversity_project_sample_attributes`
    '

thousand_genomes_sample_info_query <- '
    SELECT
      Sample AS name,
      "Illumina" AS sequencing_platform,
      Gender AS sex,
      Super_Population AS ancestry
    FROM
      `bigquery-public-data.human_genome_variants.1000_genomes_sample_info`
    '

SAMPLE_INFO_QUERIES <- list(simons = simons_sample_info_query,
                        thousand_genomes = thousand_genomes_sample_info_query,
                        platinum_genomes = thousand_genomes_sample_info_query) # Use 1000 genomes for DV platinum
```

deflaux reviewed (commented):
Nice work @johntbates !!

Main feedback is to submit the updates to the sample QC notebook to its source of truth repo and add a 'hello world' demo notebook here because I do see the value of having an example parameterized notebook right beside the two methods.

deflaux commented on programmatic_execution_of_notebooks/README.md:
```diff
@@ -29,4 +29,82 @@ Details and limitations:
+
+* [The Simons Genome Diversity Project](https://www.simonsfoundation.org/simons-genome-diversity-project/)
+* [1000 Genomes Phase 3](https://www.internationalgenome.org/category/phase-3/)
+* DeepVariant Platinum Genomes
```
Here are some links that could be added:
[DeepVariant](https://www.nature.com/articles/nbt.4235) [Platinum Genomes](https://www.ebi.ac.uk/ena/browser/view/PRJEB3381?show=component-projects)

johntbates commented:
Hi, Nicole.  I've  removed the QC demo notebook and added a "hello world" notebook that samples from a user-parameterizable normal distribution.  I've updated the README to match.

deflaux reviewed (approved):
Thanks @johntbates !
